src/node/desktop/src/main/url-utils.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * url-utils.ts
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+import http from 'http';
+import { URL } from 'url';
+import { Err } from '../core/err';
+import { logger } from '../core/logger';
+import { WaitResult, WaitTimeoutFn, waitWithTimeout } from '../core/wait-utils';
+
+/**
+ * 
+ * @param url the URL to check
+ * @returns true if is an about: url
+ */
+export function isAboutUrl(url: string): boolean {
+  return url === 'about:blank';
+}
+
+export function getAuthority(url: string): string {
+  if (isAboutUrl(url)) {
+    return url;
+  }
+
+  const targetUrl = new URL(url);
+  return `${targetUrl.protocol}//${targetUrl.host}`;
+}
+
+export function isChromeGpuUrl(url: string): boolean {
+  return url.replace(/\/$/, '') === 'chrome://gpu';
+}
+
+/**
+ * Checks the `url` if it is local
+ * 
+ * @param url the URL to check
+ * @returns true if it is local
+ */
+export function isLocalUrl(url: URL) {
+  const host = url.hostname;
+
+  return host === 'localhost' || host === '127.0.0.1' || host === '[::1]';
```

Comment:
is there a particular reason this changed from `host == "::1"` in Qt to `host === '[::1]'` here?

Reply:
Within a URL, ipv6 addresses are defined in square brackets. The URL object also returns the hostname as `[::1]`. I had to look this up when writing the unit tests.